src/nvim/tui/terminfo.c

```diff
@@ -31,7 +31,10 @@ bool terminfo_is_term_family(const char *term, const char *family)
   return tlen >= flen
     && 0 == memcmp(term, family, flen)
     // Per commentary in terminfo, minus is the only valid suffix separator.
-    && ('\0' == term[flen] || '-' == term[flen]);
+    // The screen terminfo may have a terminal name like screen.xterm. By making
```

Comment:
I am reluctant to accept such a change, if the only valid separator is `-`, than it's a first reason to avoid adding the hack, I believe screen development is on hold. At least on my distrib, screen goes by `TERM=screen` so I wonder how widespread screen.xterm is. It would be best to add some doc in the wiki or in checkhealth.

Reply:
> At least on my distrib, screen goes by `TERM=screen`

I think it probably depends on your setup. In the screen source(`termcap.c`), the part that decides the `TERM` is as follows (`tname` is the `TERM` of the host terminal. The `screenterm` is `"screen"`).

```C
        do {
                strcpy(Term, "TERM=");
                p = Term + 5;
                if (!aflag && strlen(screenterm) + strlen(tname) < MAXSTR - 1) {
                        sprintf(p, "%s.%s", screenterm, tname);
                        if (e_tgetent(buf, p) == 1)
                                break;
                }
                ...
        }
        while (0);              /* Goto free programming... */
```

Therefore, if the concatnation of `"screen."` and `TERM` of host terminal is passed to `tgetent`, it will be used as `TERM` if it succeeds. According to git's logs, I don't think it's changed since about 2005. So, I think it depends on the version of terminfo, but I think it's common to have something like `screen.xterm` set in `TERM`.

In the `man 7 term`, the root name is written as follows. Therefore, including a dot, like `screen.xterm`, seems to be a legitimate root name. But it violates the requirement of no more than seven characters.

> First,  choose  a  root  name.  The root will consist of a lower-case letter followed by up to seven lower-case letters or digits.  You need to avoid using punctuation characters in root names, because they are used and interpreted as filenames and shell meta-characters (such as !, $, *,  ?,  etc.) embedded  in them may cause odd and unhelpful behavior.  The slash (/), or any other character that may be interpreted by anyone's file system (\, $, [, ]), is especially dangerous (terminfo is platform-independent, and choosing names with special characters could someday make  life  difficult  for users  of a future port).  The dot (.) character is relatively safe as long as there is at most one per root name; some historical terminfo names use it.

Taking all of the above into consideration, I think the change is justified, as some users are actually having problems with GNU Screen.